src/framework/control.ml

```diff
@@ -677,8 +677,8 @@ struct
       Serialize.marshal marshal Fpath.(v save_run / "spec_marshal")
     );
     if get_bool "incremental.save" then (
-      Serialize.store_data marshal Serialize.AnalysisData;
-      Serialize.move_tmp_results_to_results ()
+      Serialize.Cache.(update_data AnalysisData (Obj.repr marshal));
```

Comment:
Since `Obj.repr` is inside `update_data` now, it shouldn't be necessary here any more:
```suggestion
      Serialize.Cache.(update_data AnalysisData marshal);
```
It's just that the fully generic GADT type parameter also allows passing in `Obj.t`, so the compiler wouldn't complain about this being unnecessary.

Reply:
Good catch!